### PR TITLE
String builder delete

### PR DESCRIPTION
**Pull request by Raelifin**
Builds off of https://github.com/adamant/Adamant.Compiler/pull/31

Adds an overloaded `Delete(start, [length])` method to String_Builder to clean up some lines marked as TODO in Source_File_Builder.ad

### CODE REVIEW

File: src/Program.ad [hunk 1]
```diff
@@ -565,6 +565,7 @@ public ParseArgumentsDeclaration(isMainFunction: bool, isMethod: bool) -> string
 	Expect(")");
 	let result: string = arguments.ToString();
 	// TODO implement shortening of String_Builder
+	// Max says: String_Builder has shortening (via `Delete`), but `result` is a string, not a String_Builder...
```
Comment (WalkerCodeRanger) [inline src/Program.ad]: `result` is just the `ToString()` of arguments.  The removal of the trailing comma and space should move to the `arguments` `String_Builder`.

**Review by WalkerCodeRanger — COMMENTED** [review]
Please change the method name from `Delete` to `Remove`.  As with the other pull request, you don't seem to be using translated/previous correctly.

File: src/Program.ad (hunk 1 above)
Comment (Raelifin) [reply src/Program.ad]: Done, but see my most recent comment on that code.

**Comment by WalkerCodeRanger** [comment]
This was rebased into b76ec395f2f414294c6bdd026bfa0700abffbf67 and 6db4f0b9e02202b1afe2fe587c08bfa476327124

**Pull request closed without merging**